torch/autograd/function.py

```diff
@@ -103,28 +205,24 @@ def __init__(cls, name, bases, attrs):
 
 
 # mypy doesn't understand `with_metaclass` from torch._six
-class Function(with_metaclass(FunctionMeta, _C._FunctionBase, _ContextMethodMixin, _HookMixin)):  # type: ignore[misc]
-    r"""Records operation history and defines formulas for differentiating ops.
+class Function(with_metaclass(FunctionMeta, _C._FunctionBase, Ctx, _HookMixin)):  # type: ignore[misc]
+    r"""Base class to create custom `autograd.Function`
 
-    See the Note on extending the autograd engine for more details on how to use
-    this class: https://pytorch.org/docs/stable/notes/extending.html#extending-torch-autograd
+    To create a custom `autograd.Function`, subclass this class and implement
+    the :meth:`forward` and :meth`backward` static methods.
 
-    Every operation performed on :class:`Tensor` s creates a new function
-    object, that performs the computation, and records that it happened.
-    The history is retained in the form of a DAG of functions, with edges
-    denoting data dependencies (``input <- output``). Then, when backward is
-    called, the graph is processed in the topological ordering, by calling
-    :func:`backward` methods of each :class:`Function` object, and passing
-    returned gradients on to next :class:`Function` s.
+    To ensure correctness and best performance, make sure you are calling the
+    correct methods on ``ctx`` and validating your backward function using
+    :func:`torch.autograd.gradcheck`.
 
-    Normally, the only way users interact with functions is by creating
-    subclasses and defining new operations. This is a recommended way of
-    extending torch.autograd.
+    See :ref:`extending-autograd` for more details on how to use this class.
+
+    .. note::
+        To use your custom op in your forward pass, call the ``apply`` class method.
```

Comment:
I would put this in the first intro paragraph instead of a note?
Like subclass, implement fw/bw and then use the .apply() method